src/JustSaying/AwsTools/MessageHandling/SnsPolicy.cs

```diff
@@ -1,56 +1,58 @@
-using System.Collections.Generic;
-using System.Linq;
+using System;
+using System.Text.Json;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
-using Amazon.Auth.AccessControlPolicy;
-using Amazon.Auth.AccessControlPolicy.ActionIdentifiers;
 using Amazon.SimpleNotificationService;
 using Amazon.SimpleNotificationService.Model;
 
 namespace JustSaying.AwsTools.MessageHandling
 {
-    public class SnsPolicy
+    internal static class SnsPolicy
     {
-        private readonly IReadOnlyCollection<string> _accountIds;
-
-        public SnsPolicy(IReadOnlyCollection<string> accountIds)
-        {
-            _accountIds = accountIds;
-        }
-
-        public async Task SaveAsync(string sourceArn, IAmazonSimpleNotificationService client)
+        internal static async Task SaveAsync(SnsPolicyDetails policyDetails, IAmazonSimpleNotificationService client)
         {
-            ActionIdentifier[] actions = { SNSActionIdentifiers.Subscribe };
-
-            var snsPolicy = new Policy()
-                .WithStatements(GetDefaultStatement(sourceArn))
-                .WithStatements(new Statement(Statement.StatementEffect.Allow)
-                    .WithPrincipals(_accountIds.Select(a => new Principal(a)).ToArray())
-                    .WithResources(new Resource(sourceArn))
-                    .WithActionIdentifiers(actions));
-            var attributeValue = snsPolicy.ToJson();
-            var setQueueAttributesRequest = new SetTopicAttributesRequest(sourceArn, "Policy", attributeValue);
+            var sourceAccountId = ExtractSourceAccountId(policyDetails.SourceArn);
+            var policyJson = $@"{{
+    ""Version"" : ""2012-10-17"",
+    ""Statement"" : [
+        {{
+            ""Sid"" : ""{Guid.NewGuid().ToString().Replace("-", "")}"",
```

Comment:
Do we want the SIDs to be non-deterministic? Or maybe could add a way to override the SID to something custom?

Reply:
We could remove them entirely, I was tempted to while making this change. I opted here for matching the existing behaviour.